service/worker/scheduler/workflow.go

```diff
@@ -1002,6 +1004,39 @@ func (s *scheduler) getListInfo(inWorkflowContext bool) *schedpb.ScheduleListInf
 	}
 }
 
+func (s *scheduler) updateCustomSearchAttributes(searchAttributes *commonpb.SearchAttributes) {
+	if searchAttributes == nil {
+		return
+	}
+	upsertMap := map[string]any{}
+	for key, valuePayload := range searchAttributes.GetIndexedFields() {
+		var value any
+		if err := payload.Decode(valuePayload, &value); err != nil {
+			s.logger.Error("error upserting search attributes of the scheule", "error", err)
+			continue
+		}
+		upsertMap[key] = value
+	}
+	currentSearchAttributes := workflow.GetTypedSearchAttributes(s.ctx).GetUntypedValues()
+	for key, currentValue := range currentSearchAttributes {
+		if searchattribute.IsReserved(key.GetName()) {
+			continue
+		}
+		if newValue, ok := upsertMap[key.GetName()]; !ok {
+			upsertMap[key.GetName()] = nil
+		} else if newValue == currentValue {
```

Comment:
this is going to fail on keyword list, right? do we have to use `reflect.DeepEqual` here?

Reply:
Good call. Let me fix this.